core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/SearchViewModel.kt

```diff
@@ -140,7 +140,7 @@ class SearchViewModel @Inject constructor(
 
   private fun saveSearchAndOpenItem(searchListItem: SearchListItem, openInNewTab: Boolean) {
     _effects.offer(SaveSearchToRecents(recentSearchDao, searchListItem, zimReaderContainer.id))
-    _effects.offer(OpenSearchItem(searchListItem, openInNewTab))
+    _effects.sendBlocking(OpenSearchItem(searchListItem, openInNewTab))
```

Comment:
Its necessary because we have updated the navigation lib after that offer returning false due to which search functionality breaks so its being rectify by sendBlocking. So basically its due to lib update.

Reply:
>the navigation lib after that offer returning false due to which search functionality breaks so its being rectify by sendBlocking.

Maybe it's the real bug we're encountering. We don't have to change the navigation order, which i talked in previous comment.

Reply:
hi @gouri-panda , Navigation order is being changed to fix this issue **https://github.com/kiwix/kiwix-android/issues/2886**